common/appointment/update.ts

```diff
@@ -0,0 +1,54 @@
+import { lecture as Appointment, lecture_appointmenttype_enum as AppointmentType } from '@prisma/client';
+import { prisma } from '../prisma';
+import { getUser, getStudent, User } from '../user';
+import * as Notification from '../notification';
+import { getLogger } from '../logger/logger';
+import { getAppointmentForNotification } from './util';
+import moment from 'moment';
+
+const logger = getLogger('Appointment');
+
+export async function updateAppointment(user: User, appointment: Appointment, appointmentUpdate: Partial<Appointment>) {
+    const currentDate = moment();
+    const isPastAppointment = moment(appointment.start).add(appointment.duration).isBefore(currentDate);
+
+    if (isPastAppointment) {
+        throw new Error(`Cannot update past appointment.`);
+    }
+
+    const updatedAppointment = await prisma.lecture.update({
+        where: { id: appointment.id },
+        data: { ...appointmentUpdate },
+    });
+
+    logger.info(`User(${user.userID}) updated Appointment(${appointment.id})`, { appointment, appointmentUpdate });
+
+    const sameDate = !appointmentUpdate.start || appointment.start.toISOString() === appointmentUpdate.start.toISOString();
```

Comment:
Added check whether start is actually updated

Reply:
Nice work 🎉